assets/js/googlesitekit/api/cache.js

```diff
@@ -22,12 +22,134 @@
 import { isEqual } from 'lodash';
 
 /**
- * Internal dependencies
+ * Prefix used for all Site Kit keys.
+ *
+ * Anything not using this key should not be touched by this library.
+ *
+ * @private
+ */
+export const STORAGE_KEY_PREFIX = 'googlesitekit_';
+
+const defaultOrder = [ 'localStorage', 'sessionStorage' ];
+let storageBackend;
+let storageOrder = [ ...defaultOrder	];
+/**
+ * Override the storage backend.
+ *
+ * Largely used for tests. Should not be used directly.
+ *
+ * @private
+ * @param {*} backend Backend to set for the cache.
+ * @return {void}
+ */
+export const setSelectedStorageBackend = ( backend ) => {
+	storageBackend = backend;
+};
+
+/**
+ * Override the priority of storage mechanisms.
+ *
+ * Largely used for tests. Implicitly resets the selected storage backend,
+ * causing `_getStorage` to re-run its checks for the best available
+ * storage backend.
+ *
+ * @private
+ * @param {Array} order Ordered array of storage backends to use.
+ * @return {void}
+ */
+export const setStorageOrder = ( order ) => {
+	storageOrder = [ ...order ];
+	setSelectedStorageBackend( undefined );
+};
+
+/**
+ * Reset the storage mechanism order.
+ *
+ * Largely used for tests. Implicitly resets the selected storage backend,
+ * causing `_getStorage` to re-run its checks for the best available
+ * storage backend.
+ *
+ * @private
+ * @return {void}
+ */
+export const resetDefaultStorageOrder = () => {
+	storageOrder = [ ...defaultOrder ];
+	setSelectedStorageBackend( undefined );
+};
+
+/**
+ * Detects whether browser storage is both supported and available.
+ *
+ * @private
+ * @param {string} type Browser storage to test. Should be one of `localStorage` or `sessionStorage`.
+ * @return {boolean} True if the given storage is available, false otherwise.
+ */
+export const isStorageAvailable = async ( type ) => {
+	const storage = global[ type ];
+
+	if ( ! storage ) {
+		return false;
+	}
+
+	try {
+		const x = '__storage_test__';
+
+		storage.setItem( x, x );
+		storage.removeItem( x );
+		return true;
+	} catch ( e ) {
+		return e instanceof DOMException && (
+
+			// everything except Firefox
+			22 === e.code ||
+
+			// Firefox
+			1014 === e.code ||
+
+			// test name field too, because code might not be present
+			// everything except Firefox
+			'QuotaExceededError' === e.name ||
+
+			// Firefox
+			'NS_ERROR_DOM_QUOTA_REACHED' === e.name ) &&
+
+			// acknowledge QuotaExceededError only if there's something already stored
+			0 !== storage.length;
+	}
+};
+
+/**
+ * Gets the storage object to use.
+ *
+ * @private
+ * @return {Object} Return a storage mechanism (`localStorage` or `sessionStorage`) if available; otherwise returns `null`;
```

Comment:
Would be better to omit the "return" words within the description, this is the `return` doc after all anyway.